solidago/src/solidago/pipeline/inputs.py

```diff
@@ -135,100 +155,62 @@ def __init__(self, dataset_zip: Union[str, BinaryIO]):
-                    self.comparisons[["public_username", "entity_b", "criteria"]]
-                        .rename(columns={"entity_b": "entity_id"})
-                ])  # Will contain duplicates not to be removed
+                comparisons = pd.read_csv(comparison_file, keep_default_na=False)
 
             with (zipfile.Path(zip_file) / "vouchers.csv").open(mode="rb") as vouchers_file:
                 # keep_default_na=False is required otherwise some public usernames
                 # such as "NA" are converted to float NaN.
                 self.vouchers = pd.read_csv(vouchers_file, keep_default_na=False)
 
-            with (zipfile.Path(zip_file) / "collective_criteria_scores.csv").open(mode="rb") as collective_scores_file:
-                # keep_default_na=False is required otherwise some public usernames
-                # such as "NA" are converted to float NaN.
-                self.collective_scores = pd.read_csv(collective_scores_file, keep_default_na=False)
-                # Convert video to entity_id
-                self.collective_scores["entity_id"] = self.collective_scores["video"].map(
-                    self.video_id_to_entity_id
-                )
-                self.collective_scores.drop(columns=["video"], inplace=True)
-
-                # Add a column "comparisons", as the number of comparisons made to this video
-                self.collective_scores["comparisons"] = self.collective_scores.merge(
-                    user_entity_criteria_pairs
-                        .groupby(["entity_id", "criteria"])
-                        .size()
-                        .reset_index(name="comparisons"),
-                    how="left",  # Keep all data from collective_criteria_scores in same order
-                    on=["entity_id", "criteria"],
-                )["comparisons"]
-
-                # Add a column "users", as the number of different users who have rated this video
-                self.collective_scores["users"] = self.collective_scores.merge(
-                    user_entity_criteria_pairs
-                        .groupby(["entity_id", "criteria"])
-                        .public_username
-                        .nunique()
-                        .reset_index(name="users"),
-                    how="left",  # Keep all data from collective_criteria_scores in same order
-                    on=["entity_id", "criteria"],
-                )["users"]
-
-            with (zipfile.Path(zip_file) / "individual_criteria_scores.csv").open(mode="rb") as individual_scores_file:
+            self.username_to_user_id = pd.Series(
+                data=self.users.index,
+                index=self.users["public_username"],
+            )
+
+            self.entity_id_to_video_id = pd.Series(
+                sorted(
+                    set(comparisons.video_a)
+                    | set(comparisons.video_b)
+                    | set(collective_scores.video)
```

Comment:
Including `collective_scores.video` here is necessary to make sure all videos have an `entity_id` even if they have been compare privately only (and don't appear in `comparisons`).